### PR TITLE
Replace filter implemented

### PR DESCRIPTION
**Pull request by orhanbalci**

### CODE REVIEW

**Comment by Keats** [comment]
ah sorry  I missed the `count` argument from Jinja2.
I think we can drop that and just replace everything using https://doc.rust-lang.org/std/string/struct.String.html#examples-61

Sorry for the extra work

**Comment by Keats** [comment]
also opens the discussion of mandatory args in filters.
`old` and `new` are required i'd say, defaulting to `""` is a bit weird.

Tera should probably return an error if one of them is not there but i'll change that after this PR is merged

**Comment by orhanbalci** [comment]
I can add FilterRequiredArgNotFound to TeraError and return that if you prefer

File: src/filters/string.rs [hunk 1]
```diff
@@ -116,4 +135,34 @@ mod tests {
         assert!(result.is_ok());
         assert_eq!(result.unwrap(), to_value(&4));
     }
+
+    #[test]
+    fn test_replace() {
```
Comment (Keats) [inline src/filters/string.rs]: you only need that test since we are using the built-in `replace`, you can remove the 2 other tests

Comment (orhanbalci) [inline src/filters/string.rs]: removed unnecessary tests also added capitalize filter

**Comment by Keats** [comment]
Thanks!

**Pull request merged**